src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyCommentFilter.java

```diff
@@ -399,17 +399,17 @@ public boolean isMatch(TreeWalkerAuditEvent event) {
                 final Matcher tagMatcher = tagCheckRegexp.matcher(event.getSourceName());
 
                 if (tagMatcher.find()) {
-                    match = true;
-                }
-                else if (tagMessageRegexp == null) {
-                    if (event.getModuleId() != null) {
-                        final Matcher idMatcher = tagCheckRegexp.matcher(event.getModuleId());
-                        match = idMatcher.find();
+                    if (tagMessageRegexp == null) {
+                        match = true;
+                    }
+                    else {
+                        final Matcher messageMatcher = tagMessageRegexp.matcher(event.getMessage());
+                        match = messageMatcher.find();
                     }
                 }
-                else {
-                    final Matcher messageMatcher = tagMessageRegexp.matcher(event.getMessage());
-                    match = messageMatcher.find();
+                else if (event.getModuleId() != null) {
+                    final Matcher idMatcher = tagCheckRegexp.matcher(event.getModuleId());
+                    match = idMatcher.find();
```

Comment:
If violation uses ID, and `SuppressWithNearbyCommentFilter` matches ID and is also suppressing a specific message, don't you ignore that message by not checking here, re-producing your issue in a different way?

Reply:
Oh, I will check it out, maybe that's a missing test case?

Reply:
If there are no tests for it, then yes.

Reply:
I think this is an issue with all three:
* `SuppressionCommentFilter`
* `SuppressWithNearbyCommentFilter` 
* `SuppressWithPlainTextCommentFilter`

They all use the `checkFormat` pattern to first try and match against the name of the check, and if that match fails, they reuse that pattern to match against the ID.

It seems like they are missing an `idFormat` property? Then the logic could be:
* if `checkFormat` is not null, match against `event.SourceName()`
* if `messageFormat` is not null, match against `event.getMessage()`
* if `idFormat` is not null, match against `event.getModuleId()`

I think this could then be consistent with `SuppressionFilter` which has inputs:
>* `checks` - a Regular Expression matched against the name of the check associated with an audit event. Optional as long as id or message is specified.
>* `message` - a Regular Expression matched against the message of the check associated with an audit event. Optional as long as checks or id is specified.
>* `id` - a string matched against the check id associated with an audit event. Optional as long as checks or message is specified.

Reply:
If you agree, I could make that change here - or maybe it deserves its own issue?

Anyway, as it is in this PR, I think that `isMatch` for `SuppressWithNearbyCommentFilter` implements the same logic as the other two comment filters.

Reply:
SuppressWithNearbyCommentFilter's issue which I reported should be fixed here since it is a similar issue and you are rewriting the same code. The other filters can be their own issue.

Reply:
Sounds good. Do you think an `idFormat` property should be a regular expression? Or a string, following `SuppressionFilter`?

Reply:
@esilkensen Adding a new property should be in a new issue. For this chain I am just talking about fixing the logic for matching message, weither ID or not.

Reply:
The current logic in this PR matches the logic in SuppressionCommentFilter and SuppressWithPlainTextCommentFilter, i.e. for `isMatch`
* if `checkFormat` matches the event source name
  * if `messageFormat` is null or matches event message, return true
* else if `checkFormat` matches the event ID, return true

Is this the change you would like to see in this PR?

* if `checkFormat` matches the event source name
  * if `messageFormat` is null or matches event message, return true
* else if `checkFormat` matches the event ID
  * if `messageFormat` is null or matches event message, return true

Reply:
> Is this the change you would like to see in this PR?

Yes.

Reply:
My proposal for a new property is to change it to:

* if `checkFormat` matches the event source name
  * if `messageFormat` is null or matches event message
    * if `idFormat` is null or matches event ID, return true

If that sounds good, I can create issue(s) for this and the other two comment filters.

Reply:
@rnveach, I just pushed that change, I will watch to see if CI passes and wait for your review. thanks!